independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/QuarkusClassLoader.java

```diff
@@ -239,8 +246,8 @@ public Enumeration<URL> getResources(String unsanitisedName, boolean parentAlrea
         //TODO: in theory resources could have been added in dev mode
         //but I don't think this really matters for this code path
         Set<URL> resources = new LinkedHashSet<>();
-        ClassPathElement[] providers = state.loadableResources.get(name);
-        if (providers != null) {
+        List<ClassPathElement> providers = classPathResourceIndex.getClassPathElements(name);
```

Comment:
This change will also cause an extra allocation the way we iterate over the list